### DISCUSSION TITLE
Network in the chroot environment

### DISCUSSION THREAD

**Discussion by jbhanks** [zbm-dev/zfsbootmenu#551] [Questions] [answered]
Is the chroot environment of zfsbootmenu supposed to have network? I can't even see my network interface with `ip a`. 
I tried modprobing a few things which seemed likely, but no luck. Any suggestions for getting internet working in the chroot environment?

**Comment by ahesford** [comment]
What do you mean by chroot environment?

**Reply by jbhanks** [reply]
When you press ctrl + j to "jump into chroot" from the zfsbootmenu.

**Reply by ahesford** [reply, marked as answer]
It should be possible, but if you're loading modules from within the boot environment, you may need encounter some challenges. The most likely scenario is that the kernel booted for ZBM does not match any version you have in the environment. However, if you successfully loaded modules, that doesn't seem to be the case here. There could be some firmware-loading issue, although I think it's unlikely that your regular boot process is taking special steps to load the firmware; that seems to happen more-or-less automatically when modules are loaded.

In general, if you intend to rely on networking within ZBM, you should make sure that the interface is available before you chroot into an environment. This requires either the recovery image (which contains some basic networking stuff) or a custom image that includes any necessary kernel modules.
